components/Volunteer/VolunteerSignedEvents.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import EventCard from '@/components/shared/EventCard';
+import { QueriedVolunteerEventData } from 'bookem-shared/src/types/database';
+import { fetchData } from '@/utils/utils';
+import LongEventCard from '../shared/LongEventCard';
+import { MainContainer } from '@/styles/volunteerHistory.styles';
+import SelectableLongEventCard from '../shared/SelectableLongEventCard';
+
+/**
+ * Container for all event cards
+ */
+const Container = styled.div`
+  background: ${props => props.theme.colors.BOOKEM_LIGHT_GRAY};
+  padding: 0px;
+  border-radius: 10px;
+  white-space: nowrap;
+  overflow-x: auto;
+  display: flex;
+
+  @media (max-width: 767px) {
+    padding: 0px;
+  }
+`;
+
+/**
+ * format horizontal upcoming event scroll bar on home page
+ */
+const VolunteerSignedEvents = ({
+  selectedEvent,
+  setSelectedEvent,
+}: {
+  selectedEvent: QueriedVolunteerEventData | undefined;
+  setSelectedEvent: React.Dispatch<
+    React.SetStateAction<QueriedVolunteerEventData | undefined>
+  >;
+}) => {
+  const [events, setEvents] = useState<QueriedVolunteerEventData[]>();
+  const [error, setError] = useState<Error>();
+
+  const handleEventClick = (event: QueriedVolunteerEventData) => {
+    // Update the selected event when an event is clicked
+    setSelectedEvent(event);
+  };
+
+  // Fetch upcoming events when rendered
+  useEffect(() => {
+    fetchData('/api/events/user')
+      .then(data => setEvents(data))
+      .catch(err => setError(err));
+  }, []);
```

Comment:
This /api/events/user endpoint doesn't seem to exist

Reply:
I don't seem to find this error in my code. This code is responsible for fetching and displaying the signed events in the Volunteer dashboard and it seems to work fine. There is also a endpoint for this in the project.

Reply:
Did you push all your changes? Since currently the repository doesn't have a pages/api/events/user.ts, if we do a fetch on this endpoint, it is expected to reach 404, which is what happened as I ran the code.